src/formatDate.js

```diff
@@ -50,7 +50,28 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateCollector = {};
+  const newDateFormat = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateCollector[fromFormat[i]] = dateParts[i];
+  }
+
+  if (!fromFormat.includes('YY')) {
+    dateCollector.YY = dateCollector.YYYY.slice(2);
+  } else {
+    const longYear
```

Comment:
LongYear is a redundant variable, because you can this value directly to dataCollector.YYYY